src/main/scala/org/ergoplatform/api/WalletApiRoute.scala

```diff
@@ -0,0 +1,67 @@
+package org.ergoplatform.api
+
+import akka.actor.{ActorRef, ActorRefFactory}
+import akka.http.scaladsl.server.{Directives, Route}
+import akka.pattern.ask
+import org.ergoplatform.modifiers.mempool.ErgoTransaction
+import org.ergoplatform.nodeView.ErgoReadersHolder.{GetReaders, Readers}
+import org.ergoplatform.nodeView.wallet.{ErgoWalletReader, PaymentRequest, PaymentRequestDecoder}
+import org.ergoplatform.settings.ErgoSettings
+import scorex.core.NodeViewHolder.ReceivableMessages.LocallyGeneratedTransaction
+import scorex.core.api.http.ApiError.BadRequest
+import scorex.core.api.http.ApiResponse
+import scorex.core.settings.RESTApiSettings
+
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+case class WalletApiRoute(readersHolder: ActorRef, nodeViewActorRef: ActorRef, ergoSettings: ErgoSettings)
+                         (implicit val context: ActorRefFactory) extends ErgoBaseApiRoute with ApiCodecs {
+
+  implicit val paymentRequestDecoder = new PaymentRequestDecoder(ergoSettings)
+
+  val settings: RESTApiSettings = ergoSettings.scorexSettings.restApi
+
+  private def getWallet: Future[ErgoWalletReader] = (readersHolder ? GetReaders).mapTo[Readers].map(_.w)
+
+  override val route: Route = (pathPrefix("wallet") & withCors) {
+    balancesRoute ~
+      unconfirmedBalanceRoute ~
+      sendTransactionRoute ~
+      generateTransactionRoute ~
+      generateAndSendTransactionRoute
+  }
+
+  def sendTransactionRoute: Route = (path("transaction") & post & entity(as[ErgoTransaction])) { tx =>
```

Comment:
It is copy-paste of https://github.com/ergoplatform/ergo/blob/master/src/main/scala/org/ergoplatform/api/TransactionsApiRoute.scala#L30